src/test/java/uk/gov/hmcts/reform/managecase/api/controller/NoticeOfChangeControllerTest.java

```diff
@@ -0,0 +1,171 @@
+    @AutoConfigureMockMvc(addFilters = false)
+    @ImportAutoConfiguration(TestIdamConfiguration.class)
+    static class BaseMvcTest {
+
+        @Autowired
+        protected MockMvc mockMvc;
+
+        @MockBean
+        protected NoticeOfChangeService service;
+    }
+
+    @Nested
+    @DisplayName("GET /noc/noc-questions")
+    @SuppressWarnings({"PMD.AvoidDuplicateLiterals", "PMD.JUnitTestsShouldIncludeAssert", "PMD.ExcessiveImports"})
+    class GetNoticeOfChangeQuestions extends BaseMvcTest {
+
+        @BeforeEach
+        void setUp() {
+
+        }
+
+        @Nested
+        @DisplayName("GET /noc/noc-questions")
+        class GetCaseAssignments extends BaseMvcTest {
+
+            @DisplayName("happy path test without mockMvc")
+            @Test
+            void directCallHappyPath() {
+                // created to avoid IDE warnings in controller class that function is never used
+                // ARRANGE
+                FieldType fieldType = FieldType.builder()
+                    .max(null)
+                    .min(null)
+                    .id("Number")
+                    .type("Number")
+                    .build();
+                ChallengeQuestion challengeQuestion = new ChallengeQuestion(CASE_TYPE_ID, 1,
+                                                                            "QuestionText1",
+                                                                            fieldType,
+                                                                            null,
+                                                                            "NoC",
+                                                                            ANSWER_FIELD,
+                                                                            "QuestionId1",
+                                                                            null);
+                ChallengeQuestionsResult challengeQuestionsResult = new ChallengeQuestionsResult(
+                    Arrays.asList(challengeQuestion));
+
+                given(service.getChallengeQuestions(CASE_ID)).willReturn(challengeQuestionsResult);
+
+                NoticeOfChangeController controller = new NoticeOfChangeController(service);
+
+                // ACT
+                ChallengeQuestionsResult response = controller.getNoticeOfChangeQuestions(CASE_ID);
+
+                // ASSERT
+                assertThat(response).isNotNull();
+            }
+
+            @DisplayName("should successfully get NoC questions")
+            @Test
+            void shouldGetCaseAssignmentsForAValidRequest() throws Exception {
+                FieldType fieldType = FieldType.builder()
+                    .max(null)
+                    .min(null)
+                    .id("Number")
+                    .type("Number")
+                    .build();
+                ChallengeQuestion challengeQuestion = new ChallengeQuestion(CASE_TYPE_ID, 1,
```

Comment:
If you used builder here you wouldn't need to pass in nulls ?

Reply:
done